File: easy/two_sum.py
class Solution(object):
  def twoSum(object, nums, target):
    if len(nums) == 0:
      return []
    i = 0
    while (i < len(nums)):
      first = nums[i]
      if first < target:
        if len(nums[i:]) == 0:
          return []
        j = i + 1
        while (j < len(nums)):
          second = nums[j]
          if j < target:
            if (first + second) == target:
              return [i, j]

print(twoSum([2,7,11,15], 9))

File: easy/two_sum.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution{
public:
  vector<int> two_sum(vector<int>& nums, int target)
  {
    vector<int> ret;
    if (nums.size() >= 2)
    {
      for(int i = 0; i < nums.size(); i++)
      {
        if (nums[i] < target)
        {
          for(int j = i+1; j < nums.size(); j++)
          {
            if ((nums[i] + nums[j]) == target)
            {
              ret.push_back(i);
              ret.push_back(j);
              return ret;
            }
          }
        }
      }
    }
    return ret;
  }
};

int  main()
{
  Solution mysol;
  vector<int> arg;
  arg.push_back(5);
  arg.push_back(1);
  arg.push_back(3);
  arg.push_back(2);
  vector<int> res = mysol.two_sum(arg, 7);
  vector<int>::iterator o;
  for (o = res.begin(); o < res.end(); o++)
    std::cout << (int)*o << std::endl;
  return 0;
}
